recipes/redboltz-mqtt_cpp/all/conanfile.py

```diff
@@ -1,64 +1,139 @@
-            
-    def source(self):
-        tools.get(**self.conan_data["sources"][self.version],
-                  destination=self._source_subfolder, strip_root=True)
+            check_min_cppstd(self, self._min_cppstd)
+
+        min_version = self._minimum_compiler_version.get(str(self.settings.compiler))
+        if min_version and Version(self.settings.compiler.version) < min_version:
+            raise ConanInvalidConfiguration(f"{self.ref} requires a compiler that supports at least C++{self._min_cppstd}")
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+        tc.variables["MQTT_BUILD_EXAMPLES"] = False
+        tc.variables["MQTT_BUILD_TESTS"] = False
+        tc.variables["MQTT_ALWAYS_SEND_REASON_CODE"] = self.options.mqtt_always_send_reason_code
+        tc.variables["MQTT_USE_STATIC_BOOST"] = not self.dependencies["boost"].options.get_safe("shared")
+        if self.options.with_tls:
+            tc.variables["MQTT_USE_STATIC_OPENSSL"] = not self.dependencies["openssl"].options.get_safe("shared")
+        tc.variables["MQTT_USE_TLS"] = self.options.with_tls
+        tc.variables["MQTT_USE_WS"] = self.options.use_websocket
+        if self.options.enable_std_containers:
+            tc.variables["MQTT_STD_VARIANT"] = True
+            tc.variables["MQTT_STD_OPTIONAL"] = True
+            tc.variables["MQTT_STD_STRING_VIEW"] = True
+            tc.variables["MQTT_STD_ANY"] = True
+            tc.variables["MQTT_STD_SHARED_PTR_ARRAY"] = True
+        tc.variables["MQTT_USE_LOG"] = self.options.use_logs
+        tc.variables["MQTT_DISABLE_LIBSTDCXX_TUPLE_ANY_WORKAROUND"] = self.options.with_tuple_any_workaround
+        tc.generate()
+        deps = CMakeDeps(self)
+        deps.generate()
 
     def package(self):
-        self.copy(pattern="LICENSE_1_0.txt", dst="licenses", src=self._source_subfolder)
-        self.copy(pattern="*.hpp", dst="include", src=os.path.join(self._source_subfolder, "include"))
+        copy(self, pattern="LICENSE_1_0.txt", dst=os.path.join(self.package_folder, "licenses"), src=self.source_folder)
+        copy(self, pattern="*.hpp", dst=os.path.join(self.package_folder, "include"), src=os.path.join(self.source_folder, "include"))
 
     def package_info(self):
-        self.cpp_info.set_property("cmake_file_name", "mqtt_cpp")
-        self.cpp_info.set_property("cmake_target_name", "mqtt_cpp::mqtt_cpp")
-
-        #  TODO: to remove in conan v2 once cmake_find_package_* generators removed
-        self.cpp_info.filenames["cmake_find_package"] = "mqtt_cpp"
-        self.cpp_info.filenames["cmake_find_package_multi"] = "mqtt_cpp"
-        self.cpp_info.names["cmake_find_package"] = "mqtt_cpp"
-        self.cpp_info.names["cmake_find_package_multi"] = "mqtt_cpp"
+        self.cpp_info.set_property("cmake_file_name", "mqtt_cpp_iface")
+        self.cpp_info.set_property("cmake_target_name", "mqtt_cpp_iface::mqtt_cpp_iface")
```

Comment:
```suggestion
        self.cpp_info.set_property("cmake_file_name", "mqtt_cpp_iface")
        self.cpp_info.set_property("cmake_target_name", "mqtt_cpp_iface::mqtt_cpp_iface")
```
Why did it change now? These targets were wrong or is just a new name from the latest version?